<launch>
    <arg name="uav_id" default="1"/>
    <arg name="uav_name" default="uav"/>
    <arg name="flag_printf" default="false"/>
    <arg name="flash_screen" default="true"/>
    <arg name="sim_mode" default="true"/>
    <arg name="external_source" default="2"/>
	<arg name="server" default="192.168.25.12"/>

    <arg name="vision_enable" default="true"/>
    <arg name="control_enable" default="true"/>
    <arg name="joy_enable" default="false"/>
    <arg name="ground_enable" default="false"/>
    <arg name="show_detail" default="true"/>
    <arg name="rviz_enable" default="false"/>
    <arg name="enable_rangeSensor" default="false"/>
    
    <arg name="Takeoff_height" default="1.0"/>
    <arg name="use_rc" default="true"/>

    <group if="$(arg vision_enable)">
        <include file="$(find sunray_uav_control)/launch/vision_pose_node.launch">
            <arg name="server" value="$(arg server)"/>
            <arg name="uav_id" value="$(arg uav_id)"/>
            <arg name="uav_name" value="$(arg uav_name)"/>
            <arg name="external_source" value="$(arg external_source)"/>
            <arg name="flag_printf" value="$(arg flag_printf)"/>
            <arg name="rviz_enable" value="$(arg rviz_enable)"/>
            <arg name="enable_rangeSensor" value="$(arg enable_rangeSensor)"/>
        </include>
    </group>

    <group if="$(arg control_enable)">
        <include file="$(find sunray_uav_control)/launch/sunray_control_node.launch">
            <arg name="uav_id" value="$(arg uav_id)"/>
            <arg name="uav_name" value="$(arg uav_name)"/>
            <arg name="sim_mode" value="$(arg sim_mode)"/>
            <arg name="flag_printf" value="$(arg flag_printf)"/>
            <arg name="Takeoff_height" value="$(arg Takeoff_height)"/>
            <arg name="use_rc" value="$(arg use_rc)"/>
        </include>
    </group>

    <group if="$(arg joy_enable)">
        <include file="$(find sunray_uav_control)/launch/joynode.launch">
        </include>
    </group>

    <group if="$(arg ground_enable)">
        <include file="$(find sunray_ground)/launch/ground_control.launch">
        </include>
    </group>

    <group if="$(arg show_detail)">
        <include file="$(find sunray_uav_control)/launch/show_detail.launch">
            <arg name="uav_id" value="$(arg uav_id)"/>
            <arg name="uav_name" value="$(arg uav_name)"/>
            <arg name="flash_screen" value="$(arg flash_screen)"/>
        </include>
    </group>    

</launch>